test/helpers/kubeclient_helper.rb

```diff
@@ -0,0 +1,21 @@
+module KubeclientHelper
+  def kubeclient
+    @kubeclient ||= begin
+      config = Kubeclient::Config.read(ENV["KUBECONFIG"])
+      unless config.contexts.include?("minikube")
+        raise "`minikube` context must be configured in your KUBECONFIG (#{ENV["KUBECONFIG"]})"
+      end
+
+      client = Kubeclient::Client.new(
+        config.context.api_endpoint,
+        config.context.api_version,
+        {
+          ssl_options: config.context.ssl_options,
+          auth_options: config.context.auth_options
+        }
+      )
+      client.discover
```

Comment:
It seems to get the list of resources that the API server handles and create methods for them: [code](https://github.com/abonas/kubeclient/blob/82f4059d117e31336102d41ccd99b905815b87f1/lib/kubeclient/common.rb#L94-L114). If I'm not mistaken, this makes the gem effectively able to pick up new resource types without code changes and to adapt to different server versions, but it also leads to the need to have a different client to talk to the `v1beta1` api as mentioned above.

It can also be done lazily on the first invocation of a discovery-requiring method. I copy-pasted this from Kir's work tbh, so not sure if he encountered issues with the lazy method.